documentation/writing_yaml_plans.md

```diff
@@ -538,6 +538,111 @@ steps:
 return: $hostnames.map |$hostname_result| { $hostname_result['stdout'] }
 ```
 
+## Wrapping scripts
+
+Oftentimes, you may have existing scripts that you use to perform tasks on your
+targets. While Bolt is able to run scripts on remote targets, scripts do not
+have many of the benefits of tasks and plans, such as parameterizing input or
+listing available scripts in the UI. Sometimes it might make sense to convert
+your scripts into tasks to get these benefits, but there are many instances
```

Comment:
I think it'd be nice to have a decision tree or direct comparison of when you need a task and when you need a plan. Like "You should only make your script a task if: 1...2...3. You should make your script a plan if: 1...2...3. You should leave your script just a script otherwise" or something.